runtime/src/bank.rs

```diff
@@ -3652,9 +3624,14 @@ impl Bank {
             return Err(TransactionError::UnsupportedVersion);
         }
 
-        let slot_hashes: SlotHashes = self
-            .get_cached_sysvar(&sysvar::slot_hashes::id())
-            .ok_or(TransactionError::AccountNotFound)?;
+        let slot_hashes = self
+            .sysvar_cache
+            .read()
+            .unwrap()
+            .get_slot_hashes()
+            .map_err(|_| TransactionError::AccountNotFound)?
+            .clone();
```

Comment:
Are we copying the slot_hashes data here or just the reference, looks like later we have to retake a reference to pass to `load_lookup_table_addresses`.  Should be able to just pass a reference around

Reply:
Passing a reference around would mean holding a read lock on the sysvar cache. That's actually probably fine since we shouldn't need to write to it except at the beginning or end of a block.

Reply:
Some of these sysvars are large, might save some time to not copy them

Reply:
For sure, I'll take a closer look to see if there's any risk to holding a read lock during lookups. This was one of the only things I wasn't happy with in this PR

Reply:
Might be able to pass a reference around without the lock by using an `Rc` actually

Reply:
Most, maybe all of the sysvars are bank local RO, we should be able to deser once and just read after that

Reply:
I had to `Arc` wrap because banks are shared across threads, but this avoids the cloning now.